src/components/Map/Map.jsx

```diff
@@ -1,44 +1,41 @@
 import React, { useEffect, useState } from 'react';
-import { MapContainer, TileLayer, Marker, Popup, useMap, ZoomControl } from 'react-leaflet';
+import { MapContainer, TileLayer, useMap, ZoomControl, Tooltip, Marker } from 'react-leaflet';
 import { GeoSearchControl, OpenStreetMapProvider } from 'leaflet-geosearch';
-import icons from 'leaflet-color-number-markers';
-import PopUpBox from '../PopUpBox/PopUpBox';
+import L from 'leaflet';
+import PropTypes from 'prop-types';
 
+import MarkerIcon from '../../assets/MarkerIcon.svg';
 import './Map.css';
+import { FYABackend } from '../../common/utils';
 
-const Map = () => {
-  // The activeMarker represents the current marker that is selected (i.e. which popup is shown)
-  const [activeMarker, setActiveMarker] = useState(null);
-  // This mapState variable stores the current instance of the map.
-  // This is used to fly to markers when they're clicked
+const Map = ({
+  setSelectedCountry,
+  setSelectedZipCode,
+  setSelectedBox,
+  setUpdateBoxListSwitch,
+  updateBoxListSwitch,
+}) => {
   const [mapState, setMapState] = useState(null);
+  // A list containing all unique zip codes stored in Anchor_Box
+  const [zipcodeData, setZipCodeData] = useState([]);
 
-  // This is sample data for Markers
-  // The only required pieces of data for Markers to show up are the long/lat coordinates
-  // Could use Nominatim to convert from city/state/etc name to coordinates (https://nominatim.org/release-docs/develop/api/Overview/)
-  const markerData = [
-    {
-      id: 'San Diego',
-      coordinates: [32.715736, -117.161087],
-      description: 'Sample marker at San Diego',
-      image: 'https://pngimg.com/uploads/box/box_PNG49.png',
-      number: '1',
-    },
-    {
-      id: 'Irvine',
-      coordinates: [33.6846, -117.8265],
-      description: 'Sample marker at Irvine',
-      image: 'https://pngimg.com/uploads/box/box_PNG49.png',
-      number: '2',
-    },
-    {
-      id: 'Los Angeles',
-      coordinates: [34.052235, -118.243683],
-      description: 'Sample marker at Los Angeles',
-      image: 'https://pngimg.com/uploads/box/box_PNG49.png',
-      number: '3',
-    },
-  ];
+  // Handles when a marker is clicked
+  // 1. Updates the box list with the boxes located in the zip code (in PinInformation)
+  // 2. Switches PinInformation to box list view
+  const handleMarkerClicked = markerObject => {
+    // mapState.flyTo([markerObject.latitude, markerObject.longitude], 10);
+    setSelectedCountry(markerObject.country);
+    setSelectedZipCode(markerObject.zip_code);
+    // Toggle updateBoxListSwitch, which will update update the box list in the right side bar
+    setUpdateBoxListSwitch(!updateBoxListSwitch);
+    setSelectedBox(null);
+    mapState.flyTo([markerObject.latitude, markerObject.longitude], 10);
+  };
+
+  useEffect(async () => {
+    const zipCodes = await FYABackend.get('/anchorBox/locations');
+    setZipCodeData(zipCodes.data);
+  }, []);
```

Comment:
I noticed that there was a pin on the map that was in Africa, and when I clicked on the marker the right side bar showed me the same boxes that was in the pin with the 58 boxes in Irvine. I don't know if this is a more complicated bug, but I also found that in the database one of the boxes is missing the longitude data, and as a result when you query the backend, the locations returned has a one with the latitude but the longitude is null. I attached some pics to show what is happening. Maybe just fixing that one box will fix the pins? Also this can be something to do think about in the future if there are bugs with boxes missing data.
<img width="1428" alt="Screen Shot 2022-03-29 at 4 41 46 PM" src="https://user-images.githubusercontent.com/56140648/160725178-213118ff-b7ac-4f9c-a473-1540c72cd9c3.png">
<img width="799" alt="Screen Shot 2022-03-29 at 4 51 56 PM" src="https://user-images.githubusercontent.com/56140648/160725209-b57e8892-d803-4c59-a58d-a16c4439cf30.png">

Reply:
updated getAllLocationInfo in anchorBoxService in the backend to get location info with non null coords

Reply:
I think this fix is fine, especially since we (most likely) shouldn't run into any null lat/long coordinates in the future.